src/main/java/core/mate/academy/model/Machine.java

```diff
@@ -1,6 +1,8 @@
 package core.mate.academy.model;
 
-public abstract class Machine implements Workable {
+import core.mate.academy.service.MachineProducer;
+
+public abstract class Machine implements Workable, MachineProducer<Machine> {
```

Comment:
```suggestion
public abstract class Machine implements Workable {
```

Reply:
Implements MachineProducer<Machine> It`s a bad idea, you should implement a Producer class for every type of machine in the service layer, not in model. After this re-write your logic